app/controllers/providers/address_lookups_controller.rb

```diff
@@ -1,5 +1,5 @@
 module Providers
-  class AddressLookupsController < ProviderBaseController
+  class AddressLookupsController < PreDWPCheckProviderBaseController
```

Comment:
Instead of having to change the class name, we could have it in the application_dependable.rb just like legal_aid_application_not_required! and then have that at the top of the controller like pre_dwp_check! 

it'll look like this in the end:
```
module Providers
  class AddressLookupsController < ProviderBaseController
    pre_dwp_check!
```

I did the above by coding the below:

```

    class_methods do
      def legal_aid_application_not_required!
        @legal_aid_application_not_required = true
      end

      def legal_aid_application_not_required?
        @legal_aid_application_not_required
      end

      def pre_dwp_check!
        @pre_dwp_check = true
      end

      def pre_dwp_check?
        @pre_dwp_check
      end
    end
```

Then in the LegalAidApplicationPolicy have:
( where klass = authorization_context.klass in ApplicationPolicy and klass = klass.class in AuthorizationContext )
 ```
 def authorized_passported_permissions?
    return true if @klass.pre_dwp_check?

    record.passported? ? provider.passported_permissions? : provider.non_passported_permissions?
  end
```

Reply:
What's here works well. I was just apprehensive about having to change the class name because of DWP. It feels odd to me that's why. It feels more entangled. Your thoughts?

Reply:
That's not a bad idea! Let me have a play around with it tomorrow.